python-connectors/api-connect_dataset/connector.py

```diff
@@ -49,9 +52,14 @@ def generate_rows(self, dataset_schema=None, dataset_partitioning=None,
                 record_count += len(data)
                 for row in data:
                     yield self.format_output(row, metadata)
-            else:
+            elif isinstance(data, dict):
                 record_count += 1
                 yield self.format_output(data, metadata)
+            else:
+                data = decode_csv_data(data)
```

Comment:
looks like csv format decoding is applied as a fallback when nothing else could be applied. Can't we handle this the same way as xml format, checking response header 'text/csv' ?

Reply:
This is indeed a last chance fallback, for cases where the path to data array points to a string. It is there to handle csv encapsulated in xml (or json for that matter). Therefore the header can be anything.

Reply:
oh ok I didn't have this in mind, thanks :)